src/main/scala/vectorpipe/Clip.scala

```diff
@@ -74,33 +74,39 @@ object Clip {
     centre.distanceToSegment(p1, p2) <= radius
 
   /** Naively clips Features to fit the given Extent. */
-  def byExtent[G <: Geometry, D](extent: Extent, f: Feature[G, D]): Option[Feature[Geometry, D]] = {
+  def byExtent[G <: Geometry, D](extent: Extent, f: Feature[G, D], contains: (Extent, G) => Boolean): Option[Feature[Geometry, D]] = {
```

Comment:
Hacked in a `contains` to work around having the prepared geometry contains check only knowable from the call site.